crates/compiler/gen_dev/src/generic64/storage.rs

```diff
@@ -739,16 +739,55 @@ impl<
         layout: &InLayout<'a>,
     ) {
         match layout_interner.get(*layout) {
-            Layout::Builtin(Builtin::Int(IntWidth::I64 | IntWidth::U64)) => {
-                debug_assert_eq!(to_offset % 8, 0);
-                let reg = self.load_to_general_reg(buf, sym);
-                ASM::mov_base32_reg64(buf, to_offset, reg);
-            }
-            Layout::Builtin(Builtin::Float(FloatWidth::F64)) => {
-                debug_assert_eq!(to_offset % 8, 0);
-                let reg = self.load_to_float_reg(buf, sym);
-                ASM::mov_base32_freg64(buf, to_offset, reg);
-            }
+            Layout::Builtin(builtin) => match builtin {
+                Builtin::Int(int_width) => match int_width {
+                    IntWidth::I128 | IntWidth::U128 => {
+                        // can we treat this as 2 u64's?
+                        todo!()
```

Comment:
can we "just" treat 128-bit integers (and Dec) as two 64-bit values? we'd have to get the alignment right of course, but otherwise I think that should just work?

Reply:
Yes, that should be fine as long as we align it correctly. Technically speaking if we don't pass it through a function call that must obey abi, I don't think alignment even matters.